File: AsyncTask/SampleAsyncTask/app/src/main/java/org/techtown/asynctask/MainActivity.java
package org.techtown.asynctask;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;

public class MainActivity extends AppCompatActivity {
    BackgroundTask task;
    int  value;
    ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        progressBar = findViewById(R.id.progressBar);

        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                task = new BackgroundTask();
                task.execute(); // 태스크 객체를 만들어서 실행
            }
        });

        Button button2 = findViewById(R.id.button2);
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                task.cancel(true);
            }
        });

        Button button3 = findViewById(R.id.button3);
        button3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d("Ok","ok");
                Intent intent = new Intent(getApplicationContext(),SampleThreadAnimation.class);
                startActivity(intent);
            }
        });
    }

    class BackgroundTask extends AsyncTask<Integer, Integer, Integer>{
        @Override
        protected void onPreExecute() {
            value = 0;
            progressBar.setProgress(value);
        }

        @Override
        protected Integer doInBackground(Integer... integers) {
            while (isCancelled() == false){
                value++;
                if(value >= 100){
                    break;
                }else{
                    publishProgress(value);
                }

                try{
                    Thread.sleep(10);
                } catch (InterruptedException e) { }

            }
            return value;
        }

        @Override
        protected void onProgressUpdate(Integer... values) {
            progressBar.setProgress(values[0].intValue());
        }

        @Override
        protected void onPostExecute(Integer result) {
            progressBar.setProgress(0);
        }

        protected void onCancelled(){
            progressBar.setProgress(0);
        }
    }
}

File: Kt_Fragment/app/src/main/java/org/techtown/kt_fragment/MainActivity.kt
package org.techtown.kt_fragment

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.fragment.app.FragmentTransaction

class MainActivity : AppCompatActivity() {
    lateinit var mainFragment : MainFragment
    lateinit var menuFragment : MenuFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mainFragment = MainFragment()
        menuFragment = MenuFragment()

        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.container, mainFragment).commit()
    }

    public fun onFragmentChanged(index : Int) {
        Log.d("success","getActivity")
        if (index == 0){
            supportFragmentManager.beginTransaction().replace(R.id.container,menuFragment).commit()
        }
        else {
            supportFragmentManager.beginTransaction().replace(R.id.container, mainFragment).commit()
        }
    }
}
